keras_cv/models/backbones/mix_transformer/mix_transformer_backbone.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""MiT backbone model.
+
+References:
+
+"""  # noqa: E501
+
+import copy
+
+import numpy as np
+
+from keras_cv import layers as cv_layers
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.mix_transformer.mix_transformer_backbone_presets import (  # noqa: E501
+    backbone_presets,
+)
+from keras_cv.models.backbones.mix_transformer.mix_transformer_backbone_presets import (  # noqa: E501
+    backbone_presets_with_weights,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.saving.register_keras_serializable(package="keras_cv.models")
+class MiTBackbone(Backbone):
+    def __init__(
+        self,
+        include_rescaling,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        embedding_dims=None,
+        depths=None,
+        **kwargs,
+    ):
+        drop_path_rate = 0.1
+        dpr = [x for x in np.linspace(0.0, drop_path_rate, sum(depths))]
```

Comment:
Probably not a huge deal since the presets specify `depths`, but this will raise an error because `depths=None` by default

Reply:
Good point. maybe we should specify a default depths equal to the B0 setup? 🤔

Reply:
Yeah that sgtm, or we can optionally raise an error and tell the user to either use a preset or specify `depths`.

Reply:
Removed the default value, that should signal clearly enough that depths are required to be specified or need to be built from a preset